liquibase-core/src/main/java/liquibase/Liquibase.java

```diff
@@ -259,16 +259,17 @@ public void update(Contexts contexts, LabelExpression labelExpression, boolean c
                 if (checkLiquibaseTables) {
                     checkLiquibaseTables(true, changeLog, contexts, labelExpression);
                 }
-                generateDeploymentId();
-
-                changeLog.validate(database, contexts, labelExpression);
 
                 //
                 // Let the user know that they can register for Hub
                 //
                 hubUpdater = new HubUpdater(new Date(), changeLog, database);
                 hubUpdater.register(changeLogFile);
 
+                generateDeploymentId();
```

Comment:
@filipelautert deploymentId should be generated with or without hub, then why is generateDeploymentId method gets called after hub?

Reply:
it happens that `hubUpdater.register(changeLogFile) `is calling `lockService.releaseLock()` before prompting the user if he wants to subscribe or not to Hub, and the releaseLock is clearing the deploymentId. By moving `generateDeploymentId()` to be called after Hub we don't have this problem.
I also verified and `hubUpdater.register(changeLogFile);` does not need the deploymentId .

I'm just reviewing the code as `changeLog.validate(database, contexts, labelExpression)` should still being called before hubUpdater.